sdk/core/Azure.Core/src/Shared/GuidUtilities.cs

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Buffers.Binary;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace Azure.Core.Shared;
+
+/// <summary>
+/// The purpose of this class is to provide an allocation free way to parse and write Guids on .NET Standard 2.0, taking into account
+/// endianness.
+/// </summary>
+internal static class GuidUtilities
+{
+    private const int GuidSizeInBytes = 16;
+
+    public static Guid ParseGuidBytes(ReadOnlyMemory<byte> bytes)
+    {
+        if (bytes.Length != GuidSizeInBytes)
+        {
+            return default;
+        }
+
+        ReadOnlySpan<byte> bytesSpan = bytes.Span;
+        if (BitConverter.IsLittleEndian)
+        {
+            return MemoryMarshal.Read<Guid>(bytesSpan);
+        }
+
+        // copied from https://github.com/dotnet/runtime/blob/9129083c2fc6ef32479168f0555875b54aee4dfb/src/libraries/System.Private.CoreLib/src/System/Guid.cs#L49
+        // slower path for BigEndian:
+        byte k = bytesSpan[15];  // hoist bounds checks
+        int a = BinaryPrimitives.ReadInt32LittleEndian(bytesSpan);
+        short b = BinaryPrimitives.ReadInt16LittleEndian(bytesSpan.Slice(4));
+        short c = BinaryPrimitives.ReadInt16LittleEndian(bytesSpan.Slice(6));
+        byte d = bytesSpan[8];
+        byte e = bytesSpan[9];
+        byte f = bytesSpan[10];
+        byte g = bytesSpan[11];
+        byte h = bytesSpan[12];
+        byte i = bytesSpan[13];
+        byte j = bytesSpan[14];
+
+        return new Guid(a, b, c, d, e, f, g, h, i, j, k);
+    }
+
+    public static void WriteGuidToBuffer(Guid guid, byte[] buffer)
```

Comment:
This should take Span so that I can write a GUID in a middle of an existing (longer) buffer.